### DISCUSSION TITLE
SoC graph

### DISCUSSION THREAD

**Discussion by Jadhav-Sourabh** [pybamm-team/PyBaMM#2167] [General]
How to visualize the SoC graph?

**Comment by rtimms** [comment]
This depends on how you define SoC. E.g. you could define it as the average surface or average bulk concentration in the negative electrode (normalised by the maximum concentration) in which case you could try `sim.plot(["X-averaged negative particle surface concentration", "Average negative particle concentration"])`.

If you want some other definition then you will need to extract the data from the appropriate variables.

**Reply by Jadhav-Sourabh** [reply]
I want to plot the general state of charge of the battery curve.

**Reply by Jadhav-Sourabh** [reply]
in percentage form

**Reply by Jadhav-Sourabh** [reply]
If I use "Discharge Capacity [A.h]" curve data in the opposite order as a SoC, is this ok?

**Reply by rtimms** [reply]
Yes, you could do `SoC = SoC_init - "Discharge Capacity [A.h]"/"Nominal cell capacity [A.h]"` or similar.

**Reply by Reynlab-Training** [reply]
can you please elaborate how to do that? if i do according to the above method, i'm getting an error that says 2 strings cannot be divided. I'm sorry I'm not very proficient with python. is there some fix for this? Thanks in advance

**Reply by brosaplanella** [reply]
If your solution is stored as `sol`, then you would have to write it as
```python3
SoC = SoC_init - sol["Discharge Capacity [A.h]"]/Q
```
where `Q` is the `"Nominal cell capacity [A.h]"` parameter.
